python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+            self.atspec.cmd_moveLinearStage.set(distanceFromHome=self.linear_stage)
+            await self.atspec.cmd_moveLinearStage.start()
+
+        await self.take_dark_sequence()
+
+        await self.take_bias_sequence()
+
+        await self.take_flat_sequence()
+
+        await self.take_bias_sequence()
+
+        await self.checkpoint("done")
+
+    async def take_dark_sequence(self):
+        """A coroutine to take the intended sequence of darks.
+        """
+        self.log.debug("Taking dark sequence...")
+
+        for i in range(self.n_dark):
+            await self.take_image(exp_time=self.t_dark,
+                                  shutter=False,
+                                  image_seq_name=f"dark_{i+1:04d}_{self.n_dark:04d}")
+
+    async def take_bias_sequence(self):
+        """A coroutine to take the intended sequence of bias.
+        """
+        self.log.debug("Taking bias sequence...")
+
+        for i in range(self.n_bias):
+            await self.take_image(exp_time=0.,
+                                  shutter=False,
+                                  image_seq_name=f"bias_{i+1:04d}_{self.n_bias:04d}")
+
+    async def take_flat_sequence(self):
+        """A coroutine to take the intended sequence of flats.
+        """
+        self.log.debug("Taking flat sequence...")
+
+        for i in range(len(self.flat_dn_range)):
+            for j in range(self.n_flat):
+                await self.take_image(exp_time=self.flat_base_exptime * self.flat_dn_range[i],
+                                      shutter=True,
+                                      image_seq_name=f"flat_{i+1:04d}_"
+                                                     f"{len(self.flat_dn_range):04d}_"
+                                                     f"{j+1:04d}_"
+                                                     f"{self.n_flat:04d}")
+
+    async def take_image(self, exp_time, shutter, image_seq_name):
+        """Coroutine to take images.
+
+        Parameters
+        ----------
+        exp_time : `float`
+            The exposure time for the image, in seconds.
+        shutter : `bool`
+            Should activate the shutter? (False for bias and dark)
+        image_seq_name : `str`
+            A string to identify the image.
+
+        Returns
+        -------
+
+        """
+        self.atcam.cmd_takeImages.set(numImages=1,
+                                      expTime=exp_time,
+                                      shutter=shutter,
+                                      imageSequenceName=str(image_seq_name))
+
+        timeout = self.read_out_time + self.cmd_timeout
+
+        end_readout_coro = self.atcam.evt_endReadout.next(flush=True,
+                                                          timeout=timeout)
+
+        await self.checkpoint(f"Take image {image_seq_name}")
+
+        await self.atcam.cmd_takeImages.start(timeout=timeout+exp_time)
+        await end_readout_coro
```

Comment:
I'm wondering how likely it is that this will get interleaved with other scripts that are also taking images. If that is a concern then you could make this more robust by comparing the data in the endReadout event.